core/log/multi_logger.odin

```diff
@@ -1,22 +1,53 @@
 package log
 
+import "base:runtime"
+import "core:slice"
+import "core:mem"
 
 Multi_Logger_Data :: struct {
 	loggers: []Logger,
+	allocator: runtime.Allocator,
 }
 
+@(deprecated = "Use make_multi_logger instead")
 create_multi_logger :: proc(logs: ..Logger) -> Logger {
-	data := new(Multi_Logger_Data)
-	data.loggers = make([]Logger, len(logs))
+	logger, _ := make_multi_logger(..logs, allocator = context.allocator)
+	return logger
+}
+
+/*
+Makes a new logger that will write to several other loggers.
+
+*Allocates Using Provided Allocator*
+
+Inputs:
+- logs: The loggers this logger will write to
+- allocator: (default: context.allocator)
+- loc: The caller location for debugging purposes (default: `#caller_location`)
+
+Returns:
+- res: The new multi logger 
+- res: An allocator error if one occured, `nil` otherwise 
+*/
+make_multi_logger :: proc(logs: ..Logger, allocator := context.allocator, loc := #caller_location) -> (res: Logger, err: runtime.Allocator_Error) {
+	logger_size := mem.align_forward_int(size_of(Multi_Logger_Data), align_of(Logger))
+	content_size := len(mem.slice_to_bytes(logs))
+
+	data_bytes := make([]byte, logger_size + content_size, allocator, loc) or_return
+	data := cast(^Multi_Logger_Data)raw_data(data_bytes)
+	data.loggers = slice.reinterpret([]Logger, data_bytes[logger_size:])
+	assert(len(data.loggers) == len(logs))
 	copy(data.loggers, logs)
-	return Logger{multi_logger_proc, data, Level.Debug, nil}
+	data.allocator = allocator
+
+	return Logger{multi_logger_proc, data, Level.Debug, nil}, nil
 }
 
-destroy_multi_logger :: proc(log: Logger) {
+delete_multi_logger :: proc(log: Logger) {
 	data := (^Multi_Logger_Data)(log.data)
-	delete(data.loggers)
-	free(data)
+	free(data, data.allocator)
```

Comment:
You aren't freeing the `data.loggers` slice here.

Update: Oh I see, you're doing a funny allocation dance. Maybe it's better to keep things simple and do two allocations. This isn't a place that's expected to be a performance bottleneck, so might be worth to simplify the code.

Reply:
It just means there are two places for an error to occur rather than one. It's not for a performance optimisation.

Reply:
Let's not be weird.